### DISCUSSION TITLE
How to remove namespaces?

### DISCUSSION THREAD

**Discussion by QuickMick** [socketio/socket.io#3994] [Q&A] [answered]
Hi,
i wonder if there is an official way of how to remove a dynamic namespace and disconnect each client, connected to it?? (i am speaking of this --> `const namespace = this.io.of("/" + id); `)

i was doing it like this, but it seems, that a new version of socket.io changed nsps to _nsps and to be a Map.
```
    const nsp = this.io.nsps["/" + id];
    for (let socket of Object.values(nsp.sockets)) socket.disconnect();
    nsp.removeAllListeners();
    delete this.io.nsps["/" + id];
```
i now have changed it to this:
```
    const nspId = "/" + id;
    const nsp = this.io._nsps.get(nspId);
    for (let socket of nsp.sockets.values()) socket.disconnect();
    nsp.removeAllListeners();
    this.io._nsps.delete(nspId);

```
but i guess, that is not the way, how one should do it.
so, am i missing something there? I am not able to find something in the docs.

Thanks in advance

**Comment by darrachequesne** [comment, marked as answer]
There is indeed currently no official way to remove/cleanup a namespace.

What you do should work though :+1:

**Reply by QuickMick** [reply]
thanks for the fast reply and clarification. it is just a bit scary, that nsps is now a "private" variable :D
